### ISSUE TITLE
[BUG] APIDataset._prepare_data() silently produces all-zero labels when y contains numeric values

### ISSUE THREAD

**Issue by kunal14901** [gc-os-ai/pyaptamer#387]
`APIDataset._prepare_data()` converts labels to binary via:

```python
y = torch.tensor((y == "positive").astype(int))
```

This hardcodes a string comparison against `"positive"`. When `y` is already numeric (e.g. `np.array([0, 1, 1, 0])`), the comparison `y == "positive"` evaluates to `False` for every element, producing an all-zero target tensor. No error or warning is raised.

#### To reproduce

```python
import numpy as np

y = np.array([0, 1, 1, 0])
result = (y == "positive").astype(int)
print(result)  # [0, 0, 0, 0] — all zeros regardless of actual labels
```

#### Expected behavior

Either handle both string and numeric labels, or raise a `TypeError`/`ValueError` when `y` doesn't contain expected string values.

#### Impact

A model trained on all-zero targets silently converges to always predicting the negative class. Because loss still decreases, this failure mode is hard to detect.

#### Additional context

- The type annotation is `y: np.ndarray` (generic), which doesn't restrict to string arrays
- The `load_li2014` loader docstring describes labels as `label : int`
- File: `pyaptamer/datasets/dataclasses/_api.py`, line 97

**Comment by satvshr** [comment]
Addressed [here](https://discord.com/channels/1390583813414981755/1435699734122598561/1492913522013765803)

>It's a private function with a singular input.
